shared/extension/Configuration.hpp

```diff
@@ -217,42 +223,52 @@ namespace extension {
 // NUClear configuration extension
 namespace NUClear::dsl {
     namespace operation {
+
+        /// @brief utility function to fetch the first command line argument, used to get the binary name
+        /// @returns If the first command line argument exists, it is returned. Otherwise, the empty string is returned
+        [[nodiscard]] inline std::string getFirstCommandLineArg() {
+            std::shared_ptr<const message::CommandLineArguments> args =
+                store::DataStore<message::CommandLineArguments>::get();
+            if (!args->empty()) {
+                std::vector<char> data(args->at(0).cbegin(), args->at(0).cend());
+                data.push_back('\0');
+                return std::string(data.data());
+            }
+            return std::string{};  // Empty string
+        }
+
         template <>
         struct DSLProxy<::extension::Configuration> {
+            /// @brief Binds config files, in order from least specific to most specific - default, per-robot, binary
```

Comment:
It does not bind config files, it is a bind call from nuclear. We're setting up the reaction here.

Reply:
Could you please elaborate what you mean by "it is a bind call from nuclear"? I don't-understand the distinction

Reply:
When you ask NUClear to subscribe to an event (e.g. calling `then`), represented as a type, NUClear calls the `bind` function of the type representing the event. The `bind` function is called once and is used to setup the reaction handle. This reaction handle is then returned from `then`.

`DSLProxy` is a magical type that allows you to add nuclear specific functions to already defined types.